sdk/communication/azure-communication-messages/src/main/java/com/azure/communication/messages/models/channels/WhatsAppMessageTemplateItem.java

```diff
@@ -35,10 +33,9 @@ public final class WhatsAppMessageTemplateItem extends MessageTemplateItem {
      * @param status the status value to set.
      */
     @Generated
-    @JsonCreator
-    private WhatsAppMessageTemplateItem(@JsonProperty(value = "language") String language,
-        @JsonProperty(value = "status") MessageTemplateStatus status) {
+    private WhatsAppMessageTemplateItem(String language, MessageTemplateStatus status) {
         super(language, status);
+        setKind(CommunicationMessagesChannel.WHATS_APP);
```

Comment:
`setKind` not accessible in a different pacakge

Reply:
This is a problem not related to stream-style-serialization.

Jackson would have the `setKind` method as well, and it would have the same problem.

As this case, our 0.14.x without bug fix would block service.